### ISSUE TITLE
Gem is broken on Fedora 30

### ISSUE THREAD

**Issue by thooams** [codahale/bcrypt-ruby#197]
After update to Fedora 30, here is the error message:
`BCrypt::Errors::InvalidSalt`

**Comment by dennmart** [comment]
I ran into this same issue when upgrading from Fedora 29 to Fedora 30 a few days ago. I managed to find a way to fix it on my dev environment.

Since a similar issue happened when Fedora 28 was released (see #164), I was checking to see if there were changes done to libxcrypt for this release. The only thing I found was that in Fedora 30 some [deprecated and unsafe functions were scheduled to be removed](https://fedoraproject.org/wiki/Changes/FullyRemoveDeprecatedAndUnsafeFunctionsFromLibcrypt). Their suggestion was to install the `libxcrypt-compat` package, which I did but I still had the same errors.

I then uninstalled and reinstalled the version of Ruby I was using (2.6.3, installed using rbenv) and the gem now works as before. I'm not entirely sure if the `libxcrypt-compat` package was needed here, but my guess is that it is so when Ruby is rebuilt it'll have the libraries it needs for the bcrypt gem to work.

This is just a workaround to get things running again. Hopefully there's something that can be fixed on the gem itself.
